transformer_engine/pytorch/float8_tensor.py

```diff
@@ -562,6 +562,17 @@ def __torch_dispatch__(cls, func, types, args, kwargs=None):
                 if dst._fp8_dtype == src._fp8_dtype:
                     dst._data.copy_(src._data)
                     dst._scale_inv = src._scale_inv.clone()
+                    if dst._fp8_meta is not None:
```

Comment:
Not sure if I agree with this part - if the src is already FP8, then why don't we just take the amax it is using?

Reply:
An important case is where we are loading from a checkpoint: `dst` will have an `fp8_meta` and `src` will not (pickling `fp8_meta` opens a whole bag of worms).

That said, I'm somewhat uneasy with this approach as well. Some other possible solutions:
- A fused copy-amax kernel.
- Include amax with the `Float8Tensor`, similar to `scale_inv`.
- Do not include amax updates in `Float8Tensor.copy_`, but require users to compute it externally. This isn't great since in-place operations (like the optimizer step) will no longer work with `fp8_autocast` out of the box and we can't fuse amax with the cast kernel.

Reply:
Why don't you then list out that special case explicitly, as in just go with this path if the src does not have a meta? Then you could still reuse src's amax when available but the checkpoint loading would be covered.

Reply:
Done